UEBPrep/app/controllers/sessions_controller.rb

```diff
@@ -4,9 +4,12 @@ def new
   end
 
   def create
-    users = User.where(email: params[:session][:email])
+    users = User.where(username: params[:session][:username])
     #If it's greater than one, we violated the unique email constraint
     #on the user model
+    if users.count < 1
+      users = User.where(email: params[:session][:username])
```

Comment:
well we only have one parameter coming in because it's a single field. We could do multiple fields but that just may make it more confusing.